tools/manifest/vcs.py

```diff
@@ -237,27 +241,35 @@ def check_valid(self, data):
 
     def __contains__(self, key):
         # type: (Any) -> bool
+        try:
+            key = key.decode("utf-8")
```

Comment:
Ugh, this is depressing. Can we avoid all the decode/encode stuff somehow?

Reply:
The decision seemingly was made to use bytes for file names around gitignore while I was away; I was always skeptical about that option because of the implications on the cache (having to decode/encode everywhere). The alternative is that we use unicode for filenames, then we don't need to encode/decode it for JSON.

Reply:
Yeah, I guess this is the thing where we want to handle all possible paths in a rather generic module, but then using JSON as the serialization format doesn't work. I suppose we could instead write the cache as a null-separated flat file of `path\0value\0` or similar.